src/test/java/com/redhat/cloud/policies/app/model/history/PoliciesHistoryRepositoryTest.java

```diff
@@ -40,23 +41,25 @@ class PoliciesHistoryRepositoryTest {
     @Inject
     PoliciesHistoryRepository repository;
 
-    @Test
-    void test() {
-
-        /*
-         * Some of the following history entries don't make sense from a functional perspective, but it doesn't matter.
-         * The only goal here is to test things from a technical perspective.
-         */
-        PoliciesHistoryEntry historyEntry1 = helper.createPoliciesHistoryEntry(TENANT_ID_1, ORG_ID_1, POLICY_ID_1, HOST_ID_1, HOST_NAME_1, 1L);
-        PoliciesHistoryEntry historyEntry2 = helper.createPoliciesHistoryEntry(TENANT_ID_2, ORG_ID_2, POLICY_ID_1, HOST_ID_1, HOST_NAME_1, 2L);
-        PoliciesHistoryEntry historyEntry3 = helper.createPoliciesHistoryEntry(TENANT_ID_2, ORG_ID_2, POLICY_ID_2, HOST_ID_1, HOST_NAME_1, 3L);
-        PoliciesHistoryEntry historyEntry4 = helper.createPoliciesHistoryEntry(TENANT_ID_2, ORG_ID_2, POLICY_ID_2, HOST_ID_1, HOST_NAME_2, 3L);
-        PoliciesHistoryEntry historyEntry5 = helper.createPoliciesHistoryEntry(TENANT_ID_2, ORG_ID_2, POLICY_ID_2, HOST_ID_2 + "-foo", HOST_NAME_2, 4L);
-        PoliciesHistoryEntry historyEntry6 = helper.createPoliciesHistoryEntry(TENANT_ID_2, ORG_ID_2, POLICY_ID_2, HOST_ID_2, "bar-" + HOST_NAME_2, 5L);
+    private static List<PoliciesHistoryEntry> insertedEntries;
+
+    @BeforeEach
```

Comment:
Could be replaced with a @BeforeAll annotation and we can remove the if below

Reply:
It needs to be static and so does the `helper` (`PoliciesHistoryTestHelper`). Injection of the helper, however, happens on initialized class. Any ideas how to overcome it?

Reply:
You can add `@TestInstance(TestInstance.Lifecycle.PER_CLASS)` to the class. Meaning that it will run all the tests on a single instance instead of  creating a new one per test. This allows to use the BeforeAll as a instance method.

It seems OK given that you don't have any instance fields or local state.

Reply:
Done. Thank you.